templates/ignition.yaml

```diff
@@ -0,0 +1,184 @@
+          contents: |
+            [Unit]
+            Requires=docker.socket var-lib-docker.mount
+            After=docker.socket var-lib-docker.mount
+            [Service]
+            Environment="DOCKER_OPTS=--iptables=true --storage-driver=overlay --log-opt max-size=50m --log-opt max-file=2 --live-restore --userland-proxy=false --icc=false --disable-legacy-registry=true {{index .TemplatesEnv "docker_args"}}"
+            Environment="DOCKER_OPT_BIP="
+            Environment="DOCKER_OPT_IPMASQ="
+            Environment="DOCKER_OPT_MTU="
+    - name: ensure-connectivity.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Ensures theres connectivity before starting etcd2
+        After=systemd-networkd.service
+        Wants=systemd-networkd.service
+        Before=etcd3.service
+
+        [Service]
+        Type=oneshot
+        ExecStart=/bin/sh -c 'while [[ -z $(curl -k -s {{index .TemplatesEnv "mayu_https_endpoint"}}) ]] ; do sleep 2 ; done'
+        RemainAfterExit=yes
+
+        [Install]
+        WantedBy=multi-user.target
+    - name: set-environment.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=Configures giantswarm environment
+        Wants=network-online.target
+        After=network-online.target
+
+        [Service]
+        Type=oneshot
+        ExecStartPre=/bin/sh -c "hostnamectl set-hostname machine-{{.Host.Hostname}}"
+        ExecStartPre=/bin/sh -c 'echo "SERIAL={{.Host.Serial}}" > /etc/mayu-env'
+        ExecStart=/bin/sh -c 'echo "COREOS_PRIVATE_IPV4={{.Host.InternalAddr}}" > /etc/environment'
+
+        [Install]
+        WantedBy=multi-user.target
+    - name: finish-install.service
+      enable: true
+      contents: |
+        [Unit]
+        Description=finish mayu installation
+        Before=etcd3.service
+        Wants=set-giantswarm-environment.service
+        After=set-giantswarm-environment.service
+
+        [Service]
+        Type=oneshot
+        ExecStartPre=/usr/bin/mkdir -p /opt/bin
+        ExecStartPre=/bin/sh -c 'while [[ -z $(curl -k -s {{index .TemplatesEnv "mayu_https_endpoint"}}) ]] ; do sleep 2 ; done'
+        ExecStart=/usr/bin/docker run --privileged=true -v /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt -v /etc/os-release:/etc/os-release -v /etc/mayu-env:/etc/mayu-env -v /dev:/dev -v /sys:/sys --net=host quay.io/giantswarm/mayuctl:latest boot-complete --host={{index .TemplatesEnv "mayu_api_ip"}} --port={{.MayuPort}} {{if .NoTLS}}--no-tls{{end}} --update-versions
+        RemainAfterExit=yes
+
+        [Install]
+        WantedBy=multi-user.target
+    - name: etcd3.service
+      command: start
+      enable: true
+      contents: |
+        [Unit]
+        Description=etcd
+        Requires=docker.service 
+
+        [Service]
+        Restart=always
+        RestartSec=10s
+        EnvironmentFile=/etc/environment
+        Environment=IMAGE=quay.io/coreos/etcd:v3.2.7
```

Comment:
are you sure we don't want the version in a variable? this would mean we need to create a new mayu release for a new etcd version?

Reply:
We never create a new version of mayu for changes in templates anymore, we don't use the default templates. We have our templates in `hive`. This is just an example of a template.

Reply:
ok. fair enough. but still this is very tedious to find all the versions and configurable items within the big template. having this in a smaller config files was useful imo.

Reply:
I agree, having an etcd version in mayu is probably a good step, I just don't wanna over-work this PR. AS this jsut pure mayu-config -> template stuff this can be easily done by user  or we can do the change later in the `hive`

Reply:
created issue for later tracking https://github.com/giantswarm/hive/issues/194